### PR TITLE
Add chi-squared test to categorical diff

### PR DESCRIPTION
**Pull request by andrew-yin**
The diff between categorical data now includes a chi-squared test to determine homogeneity/independence between data.

### CODE REVIEW

File: dataprofiler/profilers/categorical_column_profile.py [hunk 1]
```diff
@@ -110,6 +116,85 @@ def diff(self, other_profile, options=None):
...
+        results["df"] = df
+
+        cat_counts1 = [categories1[cat] if cat in categories1 else 0
+                       for cat in combined_cats]
+        cat_counts2 = [categories2[cat] if cat in categories2 else 0
+                       for cat in combined_cats]
+        observed = [cat_counts1, cat_counts2]
+        row_sums = [sum(cat_counts1), sum(cat_counts2)]
+        col_sums = [cat_counts1[i] + cat_counts2[i] for i in range(0, len(combined_cats))]
+        total = sum(row_sums)
```
Comment (JGSweets) [inline dataprofiler/profilers/categorical_column_profile.py:157]: We might be able to simplify this to something like:
```python
col_sums = categories1.copy()
utils.dict_merge(col_sums, categories2)

row_sums = [sample_size1, sample_size2]
total = sample_size1 + sample_size2

.
.
.

expected = dict()
chi2_statistic = 0
for key in col_sums:
    expected_cat1 = sample_size1 * col_sums.get(key, 0) / total
    expected_cat2 = sample_size2 * col_sums.get(key, 0) / total
    chi2_statistic += ... # for cat1
    chi2_statistic += ... # for cat2
```

Comment (JGSweets) [reply dataprofiler/profilers/categorical_column_profile.py:157]: would need to have sample_size as input due to static nature of the function

File: dataprofiler/profilers/categorical_column_profile.py [hunk 2]
```diff
@@ -110,6 +117,76 @@ def diff(self, other_profile, options=None):
...
+                                  categories2, n2):
+        """
+        Performs a Chi Squared test for homogeneity between two groups.
+
+        :param categories1: Categories and respective counts of the first group
+        :type categories1: dict
+        :param categories2: Categories and respective counts of the second group
+        :type categories2: dict
+        :return: Results of the chi squared test
+        :rtype: dict
```
Comment (lettergram) [inline dataprofiler/profilers/categorical_column_profile.py:131]: I only see two entries documented?

File: dataprofiler/tests/profilers/test_categorical_column_profile.py [hunk 3]
```diff
@@ -317,7 +317,10 @@ def test_categorical_diff(self):
         df_categorical = pd.Series(["y", "maybe", "y", "y", "n", "n", "maybe"])
         profile2 = CategoricalColumn(df_categorical.name)
         profile2.update(df_categorical)
-        
+
+        # chi2-statistic = sum((observed-expected)^2/expected for each category in each column)
+        # df = categories - 1
+        # p-value found through using chi2 CDF
```
Comment (lettergram) [inline dataprofiler/tests/profilers/test_categorical_column_profile.py:323]: remove?

File: dataprofiler/tests/profilers/test_categorical_column_profile.py [hunk 4]
```diff
@@ -331,6 +334,11 @@ def test_categorical_diff(self):
                     'n': 1,
                     'maybe': [None, 2]
                 }
+            },
+            'chi2-test': {
+                'chi2-statistic': 82/35,
+                'df': 2,
+                'p-value': 0.3099238764710244
```
Comment (lettergram) [inline dataprofiler/tests/profilers/test_categorical_column_profile.py:341]: How as this calculated?

File: dataprofiler/tests/profilers/test_categorical_column_profile.py (hunk 3 above)
Comment (JGSweets) [reply dataprofiler/tests/profilers/test_categorical_column_profile.py:323]: i think this comment maybe needs more clarity which will explain what's happening below.

**Review by JGSweets — APPROVED** [review]

**Review by lettergram — APPROVED** [review]

**Pull request merged**